### ISSUE TITLE
Mocking a Hive method, box.put() produces error

### ISSUE THREAD

**Issue by nazrinharris** [dart-lang/mockito#412]
So basically, I want to check whether I have passed whatever I need to pass to the HiveInterface and Box when I want to store something.

Whenever I tried to test this:
```
test(
      'should call HiveInterface and Box to cache data',
      () async {
        when(mockHiveInterface.openBox(any)).thenAnswer((_) async => mockBox);
        when(mockBox.put(0, tStoraygeUserModel))
            .thenAnswer((_) async => Future.value());
        // act
        dataSourceImpl.cacheStoraygeUser(tStoraygeUserModel);
        // assert
        verify(mockHiveInterface.openBox(STORAYGE_USER_BOX));
        verify(mockBox.put(STORAYGE_USER_ENTRY, tStoraygeUserModel));
      },
    );
```

I get this error
```
type 'Null' is not a subtype of type 'Future<void>'
MockBox.put
package:hive/…/box/box_base.dart:80
```

But this code works fine:
```
test(
      'should return StoraygeUser from StoraygeUserBox when there is one in the cache',
      () async {
        // arrange
        when(mockHiveInterface.openBox(any)).thenAnswer((_) async => mockBox);
        when(mockBox.getAt(any)).thenAnswer((_) async => tStoraygeUserModel);
        // act
        final result = await dataSourceImpl.getCachedStoraygeUser();
        // assert
        verify(mockHiveInterface.openBox(any));
        verify(mockBox.getAt(any));
        expect(result, equals(tStoraygeUserModel));
      },
    );
```

**Comment by nazrinharris** [comment]
#401

**Comment by srawlins** [comment]
@nazrinharris Are you using `@GenerateMocks`? If so, can you paste your `@GenerateMocks` annotation?

**Comment by nazrinharris** [comment]
@srawlins Yes I am. I generated the class in another file with a custom class like so:
```
@GenerateMocks([
  AuthRepository,
  AuthRepositoryImpl,
  AuthRemoteDataSource,
  AuthLocalDataSource,
  NetworkInfo,
  InternetConnectionChecker,
  HiveInterface,
  Box,
  FirebaseFirestore,
  FirebaseAuth,
])
class MockClass {}
```

Let me know if I should also put in the generated classes. (A bit long though)

**Comment by srawlins** [comment]
I suspect that `put` and `openBox` are not in the generated code, which would mean you are indeed seeing #401.

**Comment by srawlins** [comment]
This should be fixed in mockito 5.0.9.

**Comment by nazrinharris** [comment]
Yep, can confirm this is fixed in mockito 5.0.9

Thanks!
